sdk/src/sharedTest/java/io/customer/sdk/repository/ProfileRepositoryTest.kt

```diff
@@ -92,6 +94,29 @@ class ProfileRepositoryTest : BaseTest() {
         verifyNoMoreInteractions(backgroundQueueMock)
     }
 
+    @Test
+    fun identify_givenFirstTimeIdentify_givenNoDeviceTokenRegistered_expectProfileIdentifiedHookUpdateWithCorrectIdentifier() {
```

Comment:
What do you think about combining this test function with `identify_givenFirstTimeIdentify_givenDeviceTokenExists_expectIdentifyBackgroundQueue_expectDoNotDeleteToken_expectRegisterDeviceToken`? 

I am making the suggestion of combining the 2 test functions before the test environment is identical between the two. Both test functions are testing what happens when you call `identify()` when it's a first time identifying the profile *and* there is no device token registered. It would be more difficult to manage multiple tests that all have the same test environment. 

I see that you modified the test function `clearIdentify_givenNoPreviouslyIdentifiedProfile_expectIgnoreRequest_expectDontDeleteDeviceToken` to add `_expectDontUpdateHook`. This is what I am suggesting here, too. Add hooks logic to existing test function.

Reply:
Also, it would be good to add logic to the other `identify` test functions. If hooks should be called when `identify()` gets called in certain scenarios, we should also be testing that hooks *do not* get called in certain scenarios.

Reply:
> the test case doesn't get bloated but we can separate them if that happens.

I agree the test function could continue to get bloated and hard to maintain. 

If it gets too bloated to handle, I suggest we consider refactoring the `identify` function. In my experience, a bloated test function is a red flag that your code's logic is too complex.